## 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

### $p=0.8$

### $n=100$

### $k=85$

### $P(k=85)=C^{85}_{100}\cdot p^{85}\cdot q^{15}=\frac{100!}{85!15!}\cdot 0.8^{85}\cdot 0.2^{15}=0.048$


In [1]:
from math import factorial
import numpy as np

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [2]:
p = combinations(100,85)*(0.8**85)*(0.2**15)
print(f'P = {p:.3f}')

P = 0.048


### Ответ: Вероятность, что стрелок попадет в цель ровно 85 раз, равна 0.048.

## 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? 

### $\lambda=n\cdot p=5000\cdot 0.0004=2$

### $P(0)\approx \frac{\lambda^0}{0!}\cdot e^{-\lambda}\approx 2.72^{-2}\approx 0.135$

In [3]:
n = 5000
m = 0
p = 0.0004
lambda_ = n * p

P = (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)

print(f'P(0) = {P:.3f}')

P(0) = 0.135


## Какова вероятность, что перегорят ровно две?

### $P(2)\approx \frac{2^2}{2!}\cdot e^{-2}\approx \frac{4}{2}\cdot 2.72^{-2}\approx 0.27$

In [8]:
n = 5000
m = 2
p = 0.0004
lambda_ = n * p

P = (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)

print(f'P(2) = {P:.3f}')

P(2) = 0.271


### Ответ: Вероятность, что ни одна из лампочек не перегорит в первый день, равна 0.135. Вероятность, что перегорят ровно 2, равна 0.27.

## 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

### $p=0.5$

### $n=144$

### $k=70$

### $P(k=70)=C^{70}_{144}\cdot p^{70}\cdot q^{74}=\frac{144!}{70!74!}\cdot 0.5^{70}\cdot 0.5^{74}=0.063$


In [5]:
p = combinations(144,70)*(0.5**70)*(0.5**74)
print(f'P = {p:.3f}')

P = 0.063


### Ответ: Вероятность, что орел выпадет ровно 70 раз, равна 0.063.

## 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые?  

### $p=\frac{16}{22}$

### $n=4$

### $k=4$

### $P(k=4)=C^{4}_{4}\cdot p^{4}\cdot q^{0}=\frac{4!}{4!0!}\cdot (\frac{16}{22})^{4}\cdot (\frac{6}{22})^{0}=0.28$

In [6]:
p = combinations(4,4)*((16/22)**4)*((6/22)**0)
print(f'P = {p:.3f}')

P = 0.280


## Какова вероятность того, что ровно два мяча белые?

### $p=\frac{16}{22}$

### $n=4$

### $k=2$

### $P(k=2)=C^{2}_{4}\cdot p^{2}\cdot q^{2}=\frac{4!}{2!2!}\cdot (\frac{16}{22})^{2}\cdot (\frac{6}{22})^{2}=0.236$

In [7]:
p = combinations(4,2)*((16/22)**2)*((6/22)**2)
print(f'P = {p:.3f}')

P = 0.236


## Какова вероятность того, что хотя бы один мяч белый?

### $P=P(1)+P(2)+P(3)+P(4)$

### $P(1)=C^{1}_{4}\cdot p^{1}\cdot q^{3}=\frac{4!}{1!3!}\cdot (\frac{16}{22})^{1}\cdot (\frac{6}{22})^{3}=0.059$

### $P(2)=C^{2}_{4}\cdot p^{2}\cdot q^{2}=\frac{4!}{2!2!}\cdot (\frac{16}{22})^{2}\cdot (\frac{6}{22})^{2}=0.236$

### $P(3)=C^{3}_{4}\cdot p^{3}\cdot q^{1}=\frac{4!}{3!1!}\cdot (\frac{16}{22})^{3}\cdot (\frac{6}{22})^{1}=0.419$

### $P(4)=C^{4}_{4}\cdot p^{4}\cdot q^{0}=\frac{4!}{4!0!}\cdot (\frac{16}{22})^{4}\cdot (\frac{6}{22})^{0}=0.280$

In [9]:
p_1 = combinations(4,1)*((16/22)**1)*((6/22)**3)
p_2 = combinations(4,2)*((16/22)**2)*((6/22)**2)
p_3 = combinations(4,3)*((16/22)**3)*((6/22)**1)
p_4 = combinations(4,4)*((16/22)**4)*((6/22)**0)

P = p_1+p_2+p_3+p_4
print(f'P = {P:.3f}')

P = 0.994


### Ответ: Вероятность, что все вытащенные мячи белые, равна 0.280. Вероятность, что ровно два мяча белые, равна 0.236. Вероятность, что хотя бы один мяч белый, равна 0.994.